### PR TITLE
Bump mill version, split ivyDeps and compileIvyDeps

### LINKED ISSUE #69: Mill dependencies should be defined as `compileIvyDeps`
While analyzing an reported issue to Mill which only occurs in a project using this plugin, I recognized, that API dependencies to Mill (and it's dependencies like os-lib) are defined as `ivyDeps`, whereas they are supported to be defined via `compileIvyDeps`. This is because the correct runtime version is provided by Mill when the plugin is loaded into Mill's classpath and should not be included in the transitive dependency set.

### PR DESCRIPTION
**Pull request by carlosedp**
Fixes #69

### CODE REVIEW

**Review by alexarchambault — APPROVED** [review]

**Comment by alexarchambault** [comment]
Thanks for the swift PR, @carlosedp!

**Comment by carlosedp** [comment]
I've pushed tag 0.1.26 to make a new release

**Comment by carlosedp** [comment]
@alexarchambault something is not happy publishing the package...

https://github.com/alexarchambault/mill-native-image/actions/runs/6590419696/job/17907034754

File: build.sc [hunk 1]
```diff
@@ -5,7 +5,7 @@ import de.tobiasroeser.mill.vcs.version._
 import mill._, scalalib._, publish._
 import mill.scalalib.api.Util.scalaNativeBinaryVersion
 
-val millVersions       = Seq("0.9.12", "0.10.12", "0.11.0")
+val millVersions       = Seq("0.9.12", "0.10.12", "0.11.5")
```
Comment (lefou) [inline build.sc:8]: This change effectively means, you are no longer supporting Mill 0.11.0 - 0.11.4, as Mill does not guarantee any forward binary compatibilities. 

```suggestion
val millVersions       = Seq("0.9.3", "0.10.0", "0.11.0")
```

**Comment by carlosedp** [comment]
Thanks for the tip @lefou ... reverted this on #71

**Pull request merged**